python/ray/train/tests/test_backend.py

```diff
@@ -335,6 +335,139 @@ def get_resources():
     assert results == expected_results
 
 
+@pytest.mark.parametrize(
+    "worker_results",
+    [
+        (1, [[0]]),
+        (2, [[0, 1]] * 2),
+        (3, [[0]] + [[0, 1]] * 2),
+        (4, [[0, 1]] * 4),
+    ],
+)
+def test_amd_visible_devices(ray_2_node_2_gpu, worker_results):
```

Comment:
This fixture `ray_2_node_2_gpu` launch 2 nodes with 2 Nvidia GPU by default. https://github.com/ray-project/ray/blob/b572cd77a4c7be016df632c502a09bb6f7725d0b/python/ray/train/tests/conftest.py#L44-L56

You need to have some mock logic to ensure they are detected as AMD GPU.